state-chain/pallets/cf-broadcast/src/tests.rs

```diff
@@ -965,22 +1006,69 @@ fn broadcast_can_be_aborted_due_to_time_out() {
 		});
 }
 
+#[test]
+fn broadcast_timeout_works_when_external_chain_advances_multiple_blocks() {
+	new_test_ext()
+		.execute_with(|| {
+			let mut broadcast_ids = Vec::new();
+			let (broadcast_id, _) = start_mock_broadcast();
+			broadcast_ids.push(broadcast_id);
+			ready_to_abort_broadcast(broadcast_id);
+
+			let expiry = BlockHeightProvider::<MockEthereum>::get_block_height()
+				.saturating_add(crate::BroadcastTimeout::<Test, Instance1>::get());
+
+			(broadcast_ids, expiry)
+		})
+		// add start a second broadcast during the next external block
+		.then_execute_with_keep_context(|_| {
+			BlockHeightProvider::<MockEthereum>::mutate_block_height(|height| height + 1)
+		})
+		.then_execute_with(|(mut broadcast_ids, expiry)| {
+			let (broadcast_id, _) = start_mock_broadcast();
+			broadcast_ids.push(broadcast_id);
+			ready_to_abort_broadcast(broadcast_id);
+			(broadcast_ids, expiry)
+		})
+		.then_process_next_block()
+		.then_process_next_block()
+		// move external block height into the future past both timeouts
+		.then_execute_with_keep_context(|(_, expiry)| {
+			BlockHeightProvider::<MockEthereum>::set_block_height(expiry + 5)
```

Comment:
how do we know 5 is past both timeouts? It's a magic number - we could tie to the values defined more closely so it's clearer

Reply:
Speaking of magic numbers, what do you think about these?

https://github.com/chainflip-io/chainflip-backend/blob/6a9eb233cedf771bef4c65f1242f0f3e356ea776/state-chain/pallets/cf-broadcast/src/tests.rs#L516-L526

Reply:
I computed the expiry of all timeouts with `std::cmp::max` now.